### ISSUE TITLE
identifying movie on containing folder

### ISSUE THREAD

**Issue by keppo070** [barbequesauce/Watcher3#78]
**Is your feature request related to a problem? Please describe.**
I imported the following `/movies/Fantasia.1940/289989b689e14bc0a54cb3dc3b98d6af.mkv` the movie is indentified as Ariel.1988.Bluray-720P

**Describe the solution you'd like**
When unclear about filename, use foldername

**Describe alternatives you've considered**
An easy way to change the tmdb/imdb unique code for a file 

**Additional context**

**Comment by barbequesauce** [comment]
Log please... need to see how it came to that match...

**Comment by keppo070** [comment]
``INFO [2019-11-23 10:26:57,705] core.version.update_check.31: Checking for updates.
INFO [2019-11-23 10:26:57,784] core.version._update_check.265: Current local hash: 9d7a7ea9532dded021cda19791084e0590d5589c
INFO [2019-11-23 10:26:57,864] core.scheduler.create_plugin.28: Initializing scheduler plugin.
INFO [2019-11-23 10:26:57,882] cherrypy.error.error.219: [23/Nov/2019:10:26:57] ENGINE Listening for SIGTERM.
INFO [2019-11-23 10:26:57,882] cherrypy.error.error.219: [23/Nov/2019:10:26:57] ENGINE Listening for SIGHUP.
INFO [2019-11-23 10:26:57,883] cherrypy.error.error.219: [23/Nov/2019:10:26:57] ENGINE Listening for SIGUSR1.
INFO [2019-11-23 10:26:57,883] cherrypy.error.error.219: [23/Nov/2019:10:26:57] ENGINE Bus STARTING
INFO [2019-11-23 10:26:58,055] cherrypy.error.error.219: [23/Nov/2019:10:26:58] ENGINE Serving on http://0.0.0.0:9090
INFO [2019-11-23 10:26:58,056] cherrypy.error.error.219: [23/Nov/2019:10:26:58] ENGINE Bus STARTED
INFO [2019-11-23 10:26:58,264] cherrypy.error.error.219: [23/Nov/2019:10:26:58] ENGINE Started monitor thread 'Session cleanup'.
INFO [2019-11-23 10:27:57,887] CPTaskScheduler._task.250: == Executing Scheduled Task: Update Checker ==
INFO [2019-11-23 10:27:57,888] core.version.update_check.31: Checking for updates.
INFO [2019-11-23 10:27:57,900] core.version._update_check.265: Current local hash: 9d7a7ea9532dded021cda19791084e0590d5589c
INFO [2019-11-23 10:27:57,979] CPTaskScheduler._task.275: == Finished Scheduled Task: Update Checker ==
INFO [2019-11-23 10:28:12,065] core.library.scan_dir.35: Scanning /movies/Fantasia.1940 for movies.
INFO [2019-11-23 10:28:12,067] core.ajax.scan_library_directory.680: Parsing 1 directory scan results.
INFO [2019-11-23 10:28:12,067] core.ajax.scan_library_directory.682: Gathering metatadata for /movies/Fantasia.1940/289989b689e14bc0a54cb3dc3b98d6af.mkv
INFO [2019-11-23 10:28:12,067] core.library.from_file.396: Gathering metadata for /movies/Fantasia.1940/289989b689e14bc0a54cb3dc3b98d6af.mkv.
INFO [2019-11-23 10:28:12,067] core.library.parse_filename.521: Parsing directory name for movie information: Fantasia.1940.
INFO [2019-11-23 10:28:12,089] core.library.parse_filename.534: Found {'season': 89, 'episode': 14, 'container': 'mkv', 'title': '2', 'excess': '9b6c0a54cb3dc3b98d6af'} in file name.
INFO [2019-11-23 10:28:12,090] core.library.parse_media.467: Parsing codec data from file /movies/Fantasia.1940/289989b689e14bc0a54cb3dc3b98d6af.mkv.
INFO [2019-11-23 10:28:12,111] core.movieinfo.search.59: Searching TheMovieDB for 2
INFO [2019-11-23 10:28:12,111] core.movieinfo._search_tmdbid.153: Searching TheMovieDB for TMDB ID: 2.
INFO [2019-11-23 10:28:12,111] core.movieinfo._search_tmdbid.157: Searching TMDB https://api.themoviedb.org/3/movie/2?language=en-US&append_to_response=alternative_titles,external_ids,release_dates
WARNING [2019-11-23 10:28:12,613] core.movieinfo._search_tmdbid.170: TMDB returned imdbid as tt0094675
INFO [2019-11-23 10:28:12,613] core.movieinfo._search_tmdbid.153: Searching TheMovieDB for TMDB ID: 2.
INFO [2019-11-23 10:28:12,613] core.movieinfo._search_tmdbid.157: Searching TMDB https://api.themoviedb.org/3/movie/2?language=en-US&append_to_response=alternative_titles,external_ids,release_dates
WARNING [2019-11-23 10:28:13,122] core.movieinfo._search_tmdbid.170: TMDB returned imdbid as tt0094675
INFO [2019-11-23 10:28:13,122] core.ajax.scan_library_directory.699: All data found for import Ariel
INFO [2019-11-23 10:28:13,122] core.library.convert_to_db.590: Converting movie metadata to database structure for Ariel.
INFO [2019-11-23 10:28:27,911] core.ajax.import_dir.744: Adding directory scan movies to library.
INFO [2019-11-23 10:28:27,911] core.ajax.import_dir.772: Adding 1 directory scan movies to library.
INFO [2019-11-23 10:28:27,911] core.library.add_movie.867: Adding Ariel to library.
INFO [2019-11-23 10:28:27,912] core.library.convert_to_db.590: Converting movie metadata to database structure for Ariel.
INFO [2019-11-23 10:28:27,971] core.movieinfo.save.318: Downloading poster for tt0094675.
INFO [2019-11-23 10:28:27,971] core.movieinfo.save.326: Saving poster to /config/posters/tt0094675.jpg
INFO [2019-11-23 10:28:27,972] core.searchresults.generate_simulacrum.525: Creating "fake" search result for imported movie Ariel
INFO [2019-11-23 10:28:27,975] core.searchresults.score.39: Scoring 1 releases.
INFO [2019-11-23 10:28:27,975] core.searchresults.score_sources.448: Filtering resolution and size requirements.
INFO [2019-11-23 10:28:27,975] core.searchresults.score_sources.493: Keeping 1 releases.
INFO [2019-11-23 10:28:27,976] core.searchresults.fuzzy_title.363: Checking title match.
INFO [2019-11-23 10:28:27,976] core.searchresults.fuzzy_title.394: Keeping 1 releases.
INFO [2019-11-23 10:28:27,976] core.searchresults.score.118: Finished scoring releases.
INFO [2019-11-23 10:28:28,047] core.movieinfo.save.346: Poster saved to /config/posters/tt0094675.jpg

**Comment by barbequesauce** [comment]
Can I ask why you’re keeping movies with obfuscated titles in your library? 

The postprocessor can handle the functionality you’re looking for here based on the history of what was snatched and then renaming to it. As the importer doesn’t have that history, it’s getting confused by the name. 

I suppose we could have the importer (and pp engine) try and figure out if the file name is obfuscated; conceivably the importer could then try to match against the enclosing folder name - but it honestly feels like an edge case without knowing why you’re retaining the obfuscated names...

**Comment by keppo070** [comment]
I did not know it as obfuscated. I noticed it when trying to make the import directory work. There could be others.

When in importing a single movie there is an overview what movie is found. There is not a way to make a change there. I did not find a way to make a change. elsewhere.

so a solution to the import could be to ask for confirmation and a possibility to enter a different  imdb/tmdb for a manual change and ask if the obfuscated title can be renamed

**Comment by barbequesauce** [comment]
Yep... CP did that IIRC. Fair bit of work to add that in here. Any takers?

**Comment by scambra** [comment]
There is a confirmation step, with possibility of changing themoviedb ID:
![image](https://user-images.githubusercontent.com/20515/69885713-c8418c00-12de-11ea-8865-a3487d33625e.png)

Title will be changed with title from themoviedb if ID is changed

**Comment by scambra** [comment]
I have checked with a movie missing from DB, when watcher identifies movie, it doesn't allow to change TheMovieDB ID. Changing ID text with input is not enough, JS needs to be changed to send ID from input to server, and method in ajax module needs to query TheMovieDB when user changed ID.

**Comment by scambra** [comment]
Movie can be changed on import, it should be enough as work around at least
